src/mono/mono/mini/mini-arm64.c

```diff
@@ -962,6 +1021,33 @@ emit_xextract_r8 (guint8* code, int dreg, int sreg1, int sreg2)
 	return ret;
 }
 
+static guint8*
+emit_xinsert_i8_r8 (guint8* code, MonoTypeEnum type, int dreg, int src_reg, int repl_reg, int index_reg)
+{
+	guint8* ret = code;
+	gboolean is_float = is_type_float_macro (type);
+	int extra_code = 0;
+
+	if (dreg != src_reg) {
+		arm_neon_mov (ret, dreg, src_reg);
+		extra_code = 4;
+	}
+
+	arm_cbnzw (ret, index_reg, code + 12 + extra_code);
+
+	if (is_float) {
+		arm_neon_ins_e (ret, TYPE_I64, dreg, repl_reg, 0, 0);
+		arm_b (ret, code + 16 + extra_code);
```

Comment:
What is `arm_b` here?

Reply:
Is it the "branch" instruction, as in literally `b <label>`? If so, why is a branch needed here at all?

Reply:
This is a branch, the `b` instruction.

Reply:
> Is it the "branch" instruction, as in literally `b <label>`? If so, why is a branch needed here at all?

This emits the following to select one of the two `ins` instructions.
```asm
   cbnz x0, upper
   ins v0[0], d1, 0, 0
   b done
upper:
   ins v0[1], d1, 1, 0
done:
```

Reply:
Ah, I see now. This is specially handling the fallback for the non-constant case for `long`, `ulong`, and `double` since its simply `0` or `1` and avoids the need to spill/reload. 👍